File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n;
    printf("O hai! ");
    do
    {
        printf("How much change is owed?\n");
        n = get_float();
    }
    while(n<0);
    n=n*100;
    int cents=(int)roundf(n);
    int coins=0;
    while(cents>=25)
    {
        cents-=25;
        coins++;
    }
    while(cents>=10)
    {
        cents-=10;
        coins++;
    }
    while(cents>=5)
    {
        cents-=5;
        coins++;
    }
    coins = coins + cents;
    printf("%i\n", coins);
}

File: pset6/greedy.py
import cs50

print("O hai! ",end="")
while True:
    print("How much change is owed?")
    n=cs50.get_float()
    if n>0:
        break
cents = round(n*100)
coins = 0
while cents>=25:
    cents-=25
    coins+=1
while cents>=10:
    cents-=10
    coins+=1
while cents>=5:
    cents-=5
    coins+=1
coins=coins+cents
print(coins)
